src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionOperatorsTests.cs

```diff
@@ -16188,11 +16786,86 @@ static void Main()
             var comp = CreateCompilation(src, options: TestOptions.DebugExe);
             CompileAndVerify(comp, expectedOutput: "regularchecked").VerifyDiagnostics();
         }
+
+        [Fact]
+        public void CompoundAssignment_079_Consumption_ExpressionTree()
+        {
+            var src = $$$"""
+using System.Linq.Expressions;
+
+public static class Extensions1
+{
+    extension(S1 s1)
+    {
+        public static S1 operator +(S1 x, S1 y)
+        {
+            System.Console.Write("operator1");
+            return x;
+        }
+    }
+}
+
+public struct S1
+{}
+
+class Program
+{
+    static void Main()
+    {
+        Expression<System.Func<S1, S1>> ex = (s1) => s1 += s1;
+    }
+}
+""";
+
+            var comp = CreateCompilation(src, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics(
+                // (22,54): error CS0832: An expression tree may not contain an assignment operator
+                //         Expression<System.Func<S1, S1>> ex = (s1) => s1 += s1;
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsAssignment, "s1 += s1").WithLocation(22, 54)
+                );
+        }
+
+        [Fact]
+        public void CompoundAssignment_080_Consumption_ExpressionTree()
+        {
+            var src = $$$"""
+using System.Linq.Expressions;
+
+public static class Extensions1
+{
+    extension(ref S1 s1)
+    {
+        public void operator +=(S1 y)
+        {
+            System.Console.Write("operator1");
+        }
+    }
+}
+
+public struct S1
+{}
+
+class Program
+{
+    static void Main()
+    {
+        Expression<System.Func<S1, S1>> ex = (s1) => s1 += s1;
+    }
+}
+
+""" + CompilerFeatureRequiredAttribute;
+
+            var comp = CreateCompilation(src, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics(
+                // (21,54): error CS0832: An expression tree may not contain an assignment operator
+                //         Expression<System.Func<S1, S1>> ex = (s1) => s1 += s1;
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsAssignment, "s1 += s1").WithLocation(21, 54)
+                );
+        }
     }
 }
 
 // PROTOTYPE: Test unsafe and partial, IOperation/CFG , Linq expression tree, Nullable analysis
 //            Assert result of operator for basic consumption scenarios
 //            Cover ErrorCode.ERR_ExtensionDisallowsMember for conversion operators
 //            Ref safety analysis?
-//            "An extension operator may not have the same signature as a predefined operator." not yet implemented
```

Comment:
Why was this prototype comment removed?

Reply:
> Why was this prototype comment removed?

LDM decided on Monday to drop this rule.